api/src/test/java/org/bahmni/module/communication/service/impl/MessagingUtilityTest.java

```diff
@@ -0,0 +1,150 @@
+package org.bahmni.module.communication.service.impl;
+
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.util.OpenmrsUtil;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import javax.mail.Address;
+import javax.mail.Session;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+@PowerMockIgnore("javax.management.*")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Context.class)
+public class MessagingUtilityTest {
+
+    @Mock
+    private AdministrationService administrationService;
+
+    @InjectMocks
+    CommunicationServiceImpl communicationService;
+
+    private MessagingUtility messagingUtility;
+
+    @Before
+    public void setUp() {
+        PowerMockito.mockStatic(Context.class);
+        when(Context.getAdministrationService()).thenReturn(administrationService);
+        messagingUtility = new MessagingUtility();
+    }
+
+    @Test
+    public void shouldReturnSessionFromMailConfigPropertiesFile() throws IOException {
+        String TMP_FOLDER = "/tmp";
+        final File file = new File(TMP_FOLDER + "/mail-config.properties");
+        FileUtils.writeStringToFile(file, "mail.send=true\n" + "mail.transport.protocol=smtp\n"
+                + "mail.smtp.host=localhost\n" + "mail.smtp.port=25\n" + "mail.smtp.auth=false\n"
+                + "mail.smtp.starttls.enable=true\n" + "mail.from=dummy@bahmni.org\n", Charset.defaultCharset());
+        OpenmrsUtil.setApplicationDataDirectory(TMP_FOLDER);
+
+        Session session = messagingUtility.getSession();
+
+        assertEquals("localhost", session.getProperty("mail.smtp.host"));
+        file.delete();
```

Comment:
This unit test will write/delete files on each run? If the file is not deleted for some reason (e.g someone pressed control C), will the test overwrite the file in next run? 

Can we somehow write this test, so that it doesn't need to read/write a file on disk?

Reply:
Owing to the difficulty in mocking File class, I'll use the recommended [TemporaryFolder](https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html) to create the files needed to test